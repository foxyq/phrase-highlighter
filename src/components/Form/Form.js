import React from 'react';

export default function Form({ text, highlights, handleSubmit }) {
  const example = `[{startOffset: 4, endOffset: 31, color: '#d9f593', priority: 2}]`;
  return (
    <form className='form' onSubmit={handleSubmit}>
      <label htmlFor='text'>Sentence or a paragraph to be highlighted</label>
      <br />
      <textarea name='text' cols='70' rows='3' placeholder={text} />
      <label htmlFor='highlights'>Array of highlights objects</label>
      <br />
      <textarea
        name='highlights'
        cols='70'
        rows='10'
        placeholder={highlights}
      />

      <pre name='example'>{example}</pre>
      <input type='submit' value='Submit' />
    </form>
  );
}
