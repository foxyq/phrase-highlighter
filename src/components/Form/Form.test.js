import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow } from 'enzyme';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

import Form from './Form';

describe('Form component', () => {
  configure({ adapter: new Adapter() });
  it('matches snapshot', () => {
    const Component = renderer.create(<Form />).toJSON();
    expect(Component).toMatchSnapshot();
  });

  it('renders without crashing ', () => {
    const Component = renderer.create(<Form handleSubmit={() => 5} />).toJSON();
    expect(Component).toMatchSnapshot();
  });
  //

  //   it('ada', () => {
  //     const onButtonClick = sinon.spy();

  //     // expect(clickSpy.calledOnce).to.equal(true);

  //     const component = shallow(<Form handleSubmit={onButtonClick} />);
  //     component.find('input[type="submit"]').simulate('click');
  //     expect(onButtonClick.calledOnce).to.equal(true);
  //   });

  //   it('fires form submit', () => {
  //     const mockFn = jest.fn();

  //     const form = shallow(<Form onSubmit={mockFn} />);

  //     form.find('input[type="submit"]').simulate('click');
  //     form.find('input[type="submit"]').simulate('submit');

  //     expect(form.instance().onSubmit).toHaveBeenCalled();
  //     expect(mockFn).toHaveBeenCalled();
  //   });
});
