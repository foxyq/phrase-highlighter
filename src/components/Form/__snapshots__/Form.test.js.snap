// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component matches snapshot 1`] = `
<form
  className="form"
>
  <label
    htmlFor="text"
  >
    Sentence or a paragraph to be highlighted
  </label>
  <br />
  <textarea
    cols="70"
    name="text"
    rows="3"
  />
  <label
    htmlFor="highlights"
  >
    Array of highlights objects
  </label>
  <br />
  <textarea
    cols="70"
    name="highlights"
    rows="10"
  />
  <pre
    name="example"
  >
    [{startOffset: 4, endOffset: 31, color: '#d9f593', priority: 2}]
  </pre>
  <input
    type="submit"
    value="Submit"
  />
</form>
`;

exports[`Form component renders without crashing  1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <label
    htmlFor="text"
  >
    Sentence or a paragraph to be highlighted
  </label>
  <br />
  <textarea
    cols="70"
    name="text"
    rows="3"
  />
  <label
    htmlFor="highlights"
  >
    Array of highlights objects
  </label>
  <br />
  <textarea
    cols="70"
    name="highlights"
    rows="10"
  />
  <pre
    name="example"
  >
    [{startOffset: 4, endOffset: 31, color: '#d9f593', priority: 2}]
  </pre>
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
